### PR TITLE
Enable safe public logs

### PR DESCRIPTION
**Pull request by David-Henner**
## What's new in this PR?

Enabled public logging of the cryptobox events.
Sensitive information like `userId` and `deviceId` will be hashed. 

### Note

The `safePublic` method of `ZMLog` only accepts objects conforming to `SafeForLoggingStringConvertible`.
It will use the protocol's method `safeForLoggingDescription` instead of printing the object's description

### CODE REVIEW

**Comment by zenkins** [comment]
Can one of the admins verify this patch?

File: WireCryptobox.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -31,6 +31,9 @@
 		54B949941D2541350041CC55 /* TestHelper.swift in Sources */ = {isa = PBXBuildFile; fileRef = 54B949931D2541350041CC55 /* TestHelper.swift */; };
 		54C69DDA1D216A6B0060FBEC /* EncryptionContext.swift in Sources */ = {isa = PBXBuildFile; fileRef = 54C69DD91D216A6B0060FBEC /* EncryptionContext.swift */; };
 		54EA12241D2A9AA400D2CFD1 /* CryptoBoxError.swift in Sources */ = {isa = PBXBuildFile; fileRef = 54EA12231D2A9AA400D2CFD1 /* CryptoBoxError.swift */; };
+		638792502390322800FD23CF /* WireUtilities.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = 6387924F2390322800FD23CF /* WireUtilities.framework */; };
+		638792512390323000FD23CF /* WireUtilities.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = 6387924F2390322800FD23CF /* WireUtilities.framework */; };
+		638792522390328400FD23CF /* WireUtilities.framework in CopyFiles */ = {isa = PBXBuildFile; fileRef = 6387924F2390322800FD23CF /* WireUtilities.framework */; settings = {ATTRIBUTES = (CodeSignOnCopy, RemoveHeadersOnCopy, ); }; };
```
Comment (billypchan) [inline WireCryptobox.xcodeproj/project.pbxproj]: repeated lines?

Comment (David-Henner) [reply WireCryptobox.xcodeproj/project.pbxproj]: I ran into some issues with Carthage, which didn't include the framework into the project, so I had to do it manually for the main target and the test host. It seems that Xcode duplicates those lines in that case, and removing the duplicates removes the framework from `Linked frameworks and libraries`

File: WireCryptobox.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -31,6 +31,9 @@
 		54B949941D2541350041CC55 /* TestHelper.swift in Sources */ = {isa = PBXBuildFile; fileRef = 54B949931D2541350041CC55 /* TestHelper.swift */; };
 		54C69DDA1D216A6B0060FBEC /* EncryptionContext.swift in Sources */ = {isa = PBXBuildFile; fileRef = 54C69DD91D216A6B0060FBEC /* EncryptionContext.swift */; };
 		54EA12241D2A9AA400D2CFD1 /* CryptoBoxError.swift in Sources */ = {isa = PBXBuildFile; fileRef = 54EA12231D2A9AA400D2CFD1 /* CryptoBoxError.swift */; };
+		638792502390322800FD23CF /* WireUtilities.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = 6387924F2390322800FD23CF /* WireUtilities.framework */; };
+		638792512390323000FD23CF /* WireUtilities.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = 6387924F2390322800FD23CF /* WireUtilities.framework */; };
```
Comment (billypchan) [inline WireCryptobox.xcodeproj/project.pbxproj]:
```suggestion
```

**Comment by billypchan** [comment]
@zenkins release major

**Pull request merged**